test/src/test/java/org/corfudb/integration/LogReplicationSinkWriterIT.java

```diff
@@ -0,0 +1,289 @@
+        writeToActive(0, NUM_WRITES);
+
+        // At this point mapNameToMapStandby only has 2 tables. This step is also used as a barrier to indicate
+        // it's a good point to check records on Sink side
+        verifyDataOnStandby(NUM_WRITES);
+        // Verify Sink side records for Table005 and Table006 are retained, and records for Table007
+        // and Table008 are replicated.
+        // Also verify all the table's contents for snapshot sync
+        verifyTableContentsAndRecords(5, 6, 8);
+    }
+
+    /**
+     * Open tables on Source and Sink side accordingly.
+     *
+     * Tables from start to splitter - Open on both Source and Sink side, while on Sink side they will be
+     * opened with different metadata.
+     *
+     * Tables from splitter to end - Open on Source side only, after log replication verify their records
+     * exist in Sink side registry table.
+     */
+    private void prepareTablesToVerify(int start, int splitter, int end) throws Exception {
+        // From start to splitter - Open in both Source and Sink, while on Sink side they will be opened with
+        // different metadata. Verify that Sink side's records are retained.
+        for (int i = start; i <= splitter; i++) {
+            String mapName = TABLE_PREFIX + i;
+            Table<StringKey, IntValueTag, Metadata> tableActive =
+                    corfuStoreActive.openTable(NAMESPACE, mapName, StringKey.class,
+                            IntValueTag.class, Metadata.class,
+                            TableOptions.fromProtoSchema(IntValueTag.class, TableOptions.builder().build()));
+            mapNameToMapActive.put(mapName, tableActive);
+
+            Table<StringKey, IntValueTag, Metadata> tableStandby =
+                    corfuStoreStandby.openTable(NAMESPACE, mapName, StringKey.class,
+                            IntValueTag.class, Metadata.class,
+                            TableOptions.fromProtoSchema(NotFederatedTag.class, TableOptions.builder().build()));
+            mapNameToMapStandby.put(mapName, tableStandby);
+        }
+
+        // From splitter to end - Open only in Source side. Verify that their records are replicated to Sink.
+        for (int i = splitter + 1; i <= end; i++) {
+            String mapName = TABLE_PREFIX + i;
+            Table<StringKey, IntValueTag, Metadata> tableActive =
+                    corfuStoreActive.openTable(NAMESPACE, mapName, StringKey.class,
+                            IntValueTag.class, Metadata.class,
+                            TableOptions.fromProtoSchema(IntValueTag.class, TableOptions.builder().build()));
+            mapNameToMapActive.put(mapName, tableActive);
+        }
+    }
+
+    /**
+     * Verify tables records and contents after log replication.
+     */
+    private void verifyTableContentsAndRecords(int start, int splitter, int end) throws Exception {
+        // Verify Sink side records for tables before the splitter are retained, and records for the rest of
```

Comment:
nit - "Verify Sink side records for tables before the splitter are retained on the Registry table."